### PR TITLE
Use TestEnvironment::nullDevicePath instead of /dev/null/

### PR DESCRIPTION
**Pull request by davinci26**
Signed-off-by: davinci26 <sotirisnan@gmail.com>
Changes the config file to use the cross-platform `TestEnvironment::nullDevicePath()` instead of /dev/null in:

1. `rtds_integration_test.cc`
2. `cluster_integration_test.cc`
3. `redis_cluster_integration_test.cc`
4.  `mysql_integration_test.cc `
5. `postgres_integration_test.cc`
6. `rtds_integration_test.cc`

Additional Description:

The test list is flaky on Windows. I think it is related to #11793 and it will be less flaky when this change gets merged.

Risk Level: N/A
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by davinci26** [comment]
@nigriMSFT @wrowe @sunjayBhatia

**Comment by dio** [comment]
@davinci26 seems like `rtds_integration_test` is still tagged as `fails_on_windows`? Probably remove that tag to test that on CI? Thank you!

**Comment by davinci26** [comment]
@dio I added why I didn't remove the tag in the pr description. The test list is still flaky. I assume my other pr will help.

**Comment by davinci26** [comment]
@sunjayBhatia do you think we could write a format rule for this one as well?

**Comment by sunjayBhatia** [comment]
> @sunjayBhatia do you think we could write a format rule for this one as well?

yeah I think that is a good idea, especially as test cases get copied and pasted, if there are any `/dev/null`s floating around, they will keep coming back

maybe something as simple as searching for `/dev/null` in a line or even more strict `access_log_path: /dev/null`? or maybe regex like `.*: "?/dev/null"?`

https://github.com/envoyproxy/envoy/blob/2a34c76be781c8a839b74477b196f119acc51398/tools/code_format/check_format.py#L580 for examples and tests: https://github.com/envoyproxy/envoy/blob/2a34c76be781c8a839b74477b196f119acc51398/tools/code_format/check_format_test_helper.py#L149

**Comment by davinci26** [comment]
I will fix the CI (formatting) as I add the new clang-ci rule

**Comment by wrowe** [comment]
Ping? Looking forward to this merge after the format failures are fixed.

File: test/extensions/filters/network/redis_proxy/redis_proxy_integration_test.cc [hunk 1]
```diff
@@ -154,7 +155,8 @@ const std::string CONFIG_WITH_ROUTES_BASE = R"EOF(
           stat_prefix: redis_stats
           settings:
             op_timeout: 5s
-)EOF";
+)EOF",
+                                                        TestEnvironment::nullDevicePath());
```
Comment (dio) [inline test/extensions/filters/network/redis_proxy/redis_proxy_integration_test.cc:159]: This one is too far? 

Format check failed, try apply following patch to fix:
```diff
diff --git a/test/extensions/filters/network/redis_proxy/redis_proxy_integration_test.cc b/test/extensions/filters/network/redis_proxy/redis_proxy_integration_test.cc
index b67cab241..928758cd7 100644
--- a/test/extensions/filters/network/redis_proxy/redis_proxy_integration_test.cc
+++ b/test/extensions/filters/network/redis_proxy/redis_proxy_integration_test.cc
@@ -60,7 +60,7 @@ static_resources:
           settings:
             op_timeout: 5s
 )EOF",
-                                                        TestEnvironment::nullDevicePath());
+                                       TestEnvironment::nullDevicePath());

// This is a configuration with command stats enabled.
const std::string CONFIG_WITH_COMMAND_STATS = CONFIG + R"EOF(
```

Comment (davinci26) [reply test/extensions/filters/network/redis_proxy/redis_proxy_integration_test.cc:159]: I am getting really confused... I am pretty sure I had it like this in the previous commit. Look at the diff here 7cedce818faeb27c096d44cb5eb96999a8a3d715

**Comment by davinci26** [comment]
It took me more time that I am proud to admit but I finally got there. The CI is clean now.

Let me know if you have any other comments 😄

**Pull request merged**